jobmaster/system/impl.go

```diff
@@ -36,68 +39,88 @@ type Master struct {
 
 // New creates a master instance
 func New(
-	parentCtx context.Context,
 	id model.ID,
 	clients *client.Manager,
 ) *Master {
-	ctx, cancel := context.WithCancel(parentCtx)
 	return &Master{
-		ctx:     ctx,
-		cancel:  cancel,
 		id:      id,
 		clients: clients,
 
-		offExecutors:         make(chan model.ExecutorID, 100),
 		scheduleWaitingTasks: make(chan scheduleGroup, 1024),
 		scheduleRateLimit:    rate.NewLimiter(rate.Every(time.Second), 1),
 
-		execTasks:    make(map[model.ExecutorID][]*model.Task),
-		runningTasks: make(map[model.ID]*Task),
+		runningTasks: make(map[model.ID]*model.Task),
 	}
 }
 
 func (m *Master) Cancel() {
```

Comment:
what's the relationship between the master  and jobmaster

Reply:
"Master" is the base class of "JobMaster"